src/modules/services/hooks.js

```diff
@@ -58,3 +61,23 @@ export const useWorkflows = () => {
         return { workflowsByType, workflowsLoading };
     }, [serviceWorkflows, workflowsLoading]);
 };
+
+
+export const useMetadataDependentData = () => {
```

Comment:
i think it would be better to have two hooks which actually return the redux state of the corresponding data; `useExtraPropertiesSchemaTypes` and `useDiscoverySchema`, which themselves can call the fetch if katsu is ready. this creates a cleaner path from data-dependency to the fetch action.